### PR TITLE
add Matrix Me

### PR DESCRIPTION
**Pull request by yulei**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/matrix/me/rules.mk [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = atmel-dfu
```
Comment (drashna) [inline keyboards/matrix/me/rules.mk:12]:
```suggestion
```

File: keyboards/matrix/me/rules.mk [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite 	# Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/matrix/me/rules.mk:17]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic
```

File: keyboards/matrix/me/rules.mk [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
```
Comment (drashna) [inline keyboards/matrix/me/rules.mk:4]:
```suggestion
# Bootloader selection
BOOTLOADER = atmel-dfu 
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/matrix/me/config.h [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+/* disable debug print */
+//#define NO_DEBUG
+
+/* disable print */
+//#define NO_PRINT
+
+//rgb light setting
+#define RGBLED_NUM          4
+#define RGB_DI_PIN          E2
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/matrix/me/config.h:52]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/matrix/me/rules.mk [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no			# USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+RGBLIGHT_ENABLE = yes 		# Use RGB underglow light
```
Comment (fauxpark) [inline keyboards/matrix/me/rules.mk:24]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/matrix/me/readme.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# Matrix ME
+
+A TKL keyboard with top gasket
+
+Keyboard Maintainer: [astro](https://github.com/yulei)  
+Hardware Supported: Matrix ME keyboard
+Hardware Availability: GB
```
Comment (fauxpark) [inline keyboards/matrix/me/readme.md:7]:
```suggestion
* Keyboard Maintainer: [astro](https://github.com/yulei)
* Hardware Supported: Matrix ME keyboard
* Hardware Availability: GB
```

File: keyboards/matrix/me/info.json [hunk 7]
```diff
@@ -0,0 +1,15 @@
+{
+  "keyboard_name": "Matrix ME",
+  "url": "",
+  "maintainer": "qmk",
+  "width": 18.25,
+  "height": 6.5,
```
Comment (fauxpark) [inline keyboards/matrix/me/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Comment by wgehalo** [comment]
@yulei Forgive my ignorance, is this update for the stock PCB that comes with the Matrix ME?

**Pull request merged**